src/Esfa.Vacancy.Register.Infrastructure/Caching/AzureRedisCacheService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Threading.Tasks;
+using Esfa.Vacancy.Register.Domain.Interfaces;
+using Esfa.Vacancy.Register.Infrastructure.Settings;
+using Newtonsoft.Json;
+using SFA.DAS.NLog.Logger;
+using StackExchange.Redis;
+
+namespace Esfa.Vacancy.Register.Infrastructure.Caching
+{
+
+    public class AzureRedisCacheService : ICacheService
+    {
+        private readonly Lazy<ConnectionMultiplexer> _lazyConnection;
+        private readonly ILog _logger;
+
+        public AzureRedisCacheService(IProvideSettings settings, ILog logger)
+        {
+            _logger = logger;
+            var cacheConnection = settings.GetSetting(ApplicationSettingKeyConstants.CacheConnection);
+            _lazyConnection = new Lazy<ConnectionMultiplexer>(() => ConnectionMultiplexer.Connect(cacheConnection));
+        }
+
+        private ConnectionMultiplexer Connection => _lazyConnection.Value;
+
+        public async Task<T> CacheAsideAsync<T>(string key, Func<Task<T>> actionAsync, TimeSpan timeSpan)
+        {
+            var cache = Connection.GetDatabase();
+
+            var cachedValue = await cache.StringGetAsync(key);
+
+            T result;
+
+            if (cachedValue.HasValue)
+            {
+                result = JsonConvert.DeserializeObject<T>(cachedValue);
+                _logger.Info($"Redis read key={key}");
```

Comment:
I have a feeling this might flood the logs. Maybe we just depend on the logging of the caller to this and have the below logging line for when it is not reading from the cache. Maybe we can discuss this.

Reply:
Agree. The ticket mentions logging on read and population but doing so on read I think is overkill. Lets discuss

Reply:
I think this logging is necessary, this is the only way we will be able to tell if the values are being read from Cache or real source in case we ever have to debug. There is a guide line here https://skillsfundingagency.atlassian.net/wiki/spaces/DAS/pages/46301186/Logging+levels